dashboard/modules/job/job_head.py

```diff
@@ -204,6 +212,80 @@ async def list_jobs(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    @routes.get("/api/all_jobs/")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
+    async def list_jobs(self, req: Request) -> Response:
+        driver_jobs, submission_job_drivers = await self._get_driver_jobs()
+        submission_jobs = self._job_manager.list_jobs()
+        for job_id, job in submission_jobs.items():
+            job.driver = submission_job_drivers.get(job_id)
+        return Response(
+            text=json.dumps(
+                [
+                    *[
+                        {
+                            **dataclasses.asdict(job_info),
+                            "id": job_id,
+                            "type": "submission",
+                        }
+                        for job_id, job_info in submission_jobs.items()
+                    ],
+                    *[
+                        {**dataclasses.asdict(job_info), "id": job_id, "type": "driver"}
+                        for job_id, job_info in driver_jobs.items()
+                    ],
+                ]
+            ),
+            content_type="application/json",
+        )
+
+    async def _get_driver_jobs(self):
+        """
+        Returns a tuple of dictionaries related to drivers.
```

Comment:
Can you follow https://docs.ray.io/en/latest/ray-contribute/getting-involved.html#code-style for the docstring?